### PR TITLE
Fix for notifications migration on postgresql and mssql

### LINKED ISSUE #3993: PostgreSQL changelog issue
Looks like upgrade doesn't work well on PostgreSQL:

https://www.traccar.org/forums/topic/versiya-40/page/2/#post-33622

@Abyss777, any ideas how to solve it?

### PR DESCRIPTION
**Pull request by Abyss777**
Made separate migrations for h2, mysql, mssql(`common`) and postgresql (`pg`).
Split on `pre`, "main" and `post`.
Added preconditions in case somebody already successfully migrated to 4.0 for future.

It might looks a bit complicated but I hope be universal.

fix #3993

### CODE REVIEW

File: schema/changelog-4.0.xml [hunk 1]
```diff
@@ -94,8 +112,69 @@
...
+    </update>
+
+    <update tableName="tc_notifications">
+        <column name="notificators" value="sms" />
+        <where>web = FALSE AND mail = FALSE AND sms = TRUE</where>
+    </update>
+
+  </changeSet>
+
+  <changeSet author="author" id="changelog-4.0-post">
```
Comment (tananaev) [inline schema/changelog-4.0.xml]: Can we just name this one `changelog-4.0`? So that we don't have one extra record?

Comment (tananaev) [reply schema/changelog-4.0.xml]: It would also mean that we don't need a check.

**Comment by Abyss777** [comment]
You are right, renamed it.

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**